dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoResource.java

```diff
@@ -1004,4 +1007,10 @@ private boolean checkIfUserOwns(final Tool tool, final long userId) {
         }
         return false;
     }
+
+    public void checkNotHosted(Tool tool) {
+        if (tool.getMode() == ToolMode.HOSTED) {
+            throw new WebApplicationException("Cannot modify hosted entries this way", HttpStatus.SC_BAD_REQUEST);
```

Comment:
What the difference between CustomWebApplicationException and WebApplicationException?

Reply:
Not sure but I'll change to custom since that seems like what we use.